apps/aehyperchains/src/aehc_consensus_hyperchains.erl

```diff
@@ -1039,9 +1064,83 @@ protocol_staking_contract_call(Trees0, TxEnv, Query) ->
         {error, _What} = Err -> Err
     end.
 
+
+%% Implementation
+
+-spec load_hc_staking_contract() -> ok | no_return().
+load_hc_staking_contract() ->
+    P = aec_fork_block_settings:hc_staking_contract_file(),
+    case filelib:is_file(P) of
+        true -> ok;
+        false ->
+            aec_consensus:config_assertion_failed(
+                "Hyperchains staking contract not present.",
+                " Terminating the node - file not found ~p", [P])
+    end,
+    ldebug("Loading hyperchains staking contract"),
+    {ok, Data} = file:read_file(P),
+    try
+        #{
+            <<"bytecode">> := B,
+            <<"aci">> := JText
+        } = jsx_decode(Data),
+        CompiledACI = aeaci_aci:from_string(JText, #{backend => fate}),
+        {contract_bytearray, Bytecode} = decode(B),
+        persistent_term:put(?STAKING_CONTRACT, #{bytecode => Bytecode, aci => CompiledACI})
+    catch E:R:Stacktrace ->
+        aec_consensus:config_assertion_failed(
+            "Hyperchains staking contract is malformed.",
+            " Terminating the node - ~p ~p ~p", [E, R, Stacktrace])
+    end.
+
 %% What consensus shall we use in case the activation criteria of the hyperchain consensus weren't meet?
 %% TODO: customize
-fallback_consensus() ->
-    aec_consensus_bitcoin_ng.
+fallback_consensus() -> aec_consensus_bitcoin_ng.
 
 hc_genesis_version() -> ?HC_GENESIS_VERSION.
+
+
+%% Helpers
+
+-spec apply_fallback(atom()) -> term().
+apply_fallback(Function) -> apply_fallback(Function, []).
+
+-spec apply_fallback(atom(), [term()]) -> term().
+apply_fallback(Function, Args) -> apply(fallback_consensus(), Function, Args).
+
+-spec jsx_decode(jsx:json_text()) -> map() | no_return().
+jsx_decode(Data) ->
+    case jsx:decode(Data, [{return_maps, true}, {labels, binary}]) of
+        Map when is_map(Map) ->
+            Map;
+        [Map] when is_map(Map) ->
+            Map
+    end.
+
+-spec encode_contract_pubkey(pubkey()) -> binary().
+encode_contract_pubkey(Pubkey) -> aeser_api_encoder:encode(contract_pubkey, Pubkey).
```

Comment:
I think introducing one line helpers like this makes the code actually less readable.

Reply:
> I think introducing one line helpers like this makes the code actually less readable.

Well, IMO doing something like `M=module, M:f(a)` everywhere is way worse than abstracting m:f:a calls with `apply`, but it's your call. Revert now or pause to weigh it a bit more?